src/expr/src/relation/mod.rs

```diff
@@ -507,17 +507,25 @@ impl RelationExpr {
 
     /// Retains only the columns specified by `output`.
     pub fn project(self, outputs: Vec<usize>) -> Self {
-        RelationExpr::Project {
-            input: Box::new(self),
-            outputs,
+        if outputs.len() == self.arity() && outputs.iter().enumerate().all(|(i, j)| i == *j) {
+            self
+        } else {
```

Comment:
This seems like a much smarter place to put this than in-line in all of the analyses themselves!